helios-system-tests/src/main/java/com/spotify/helios/system/SystemTestBase.java

```diff
@@ -602,7 +615,34 @@ protected AgentMain startDefaultAgent(final String host, final String... args)
         dockerPortRange.upperEndpoint()
     );
     argsList.addAll(asList(args));
-    return startAgent(argsList.toArray(new String[argsList.size()]));
+    return startAgentWithArgs(argsList);
+  }
+
+  protected AgentMain startAgentWithArgs(final List<String> args) throws Exception {
+    final AgentMain agentMain = startAgent(args.toArray(new String[args.size()]));
+    waitForAgentToBeFullyUp();
```

Comment:
I think this will fix test failures like DeregisterTest.testJobsArePreservedWhenReregistering()` here https://circleci.com/gh/spotify/helios/3795.